api/app/references/repository.py

```diff
@@ -0,0 +1,28 @@
+from app import db
+from app.references.models import ReferenceRequest
+from app.events.models import Event
+import uuid
+
+
+class ReferenceRequestRepository():
+    
+    @staticmethod
+    def get_by_id(reference_request_id):
+        return db.session.query(ReferenceRequest).get(reference_request_id)
+
+    @staticmethod
+    def create(reference_request):
+        reference_request.set_token(str(uuid.uuid4()))
+        db.session.add(reference_request)
+        db.session.commit()
+    
+    @staticmethod
+    def update(reference_request):
+        db.session.add(reference_request)
```

Comment:
Are we sure this doesn't add another row to the table?

Reply:
Well, that depends on the entity passed to the function. If the entity has no id, it will definitely create a new row but if there is id and exist it will merge. 
But I also added an assertion on POST request for that.